src/ngraph_builder.cc

```diff
@@ -612,6 +612,45 @@ static tf::Status TranslateBiasAddOp(const tf::Node* op,
   return tf::Status::OK();
 }
 
+static tf::Status TranslateBiasAddGradOp(const tf::Node* op,
+                                     Builder::OpMap& ng_op_map) {
+  shared_ptr<ng::Node> ng_input;
+  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input));
+
+  std::string tf_data_format;
+  if (tf::GetNodeAttr(op->attrs(), "data_format", &tf_data_format) !=
+      tf::Status::OK()) {
+    tf_data_format = "NHWC";
+  }
+
+  if (tf_data_format != "NHWC" && tf_data_format != "NCHW") {
+    return tf::errors::InvalidArgument(
+        "BiasAddGrad data format is neither NHWC nor NCHW");
+  }
+
+  auto ng_input_shape = ng_input->get_shape();
+  if (ng_input_shape.size() != 4) {
+    return tf::errors::InvalidArgument(
+        "BiasAddGrad argument does not have 4 dimensions");
```

Comment:
The `BiasAddGrad` docs at https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/bias-add-grad indicate that `out_backprop` can have any number of dimensions. If 4D is all we need for the moment then that's fine, but let's register that restriction in the confirmation pass. If we can support n-dimensional inputs, that's even better.

Reply:
We can support >=2D for `NHWC` data format but Tensorflow only supports 4D for `NCHW`.  Should we register this restriction in the confirmation pass? https://github.com/tensorflow/tensorflow/blob/f202958ee2d5177a474e3d107fdbf0c83174d099/tensorflow/core/kernels/bias_op.cc#L249